manifests/init.pp

```diff
@@ -445,153 +445,130 @@
-  $log_dir                        = '/var/log/sensu',
-  $dashboard                      = false,
-  $init_stop_max_wait             = 10,
-  $gem_install_options            = undef,
-  $hasrestart                     = true,
-  $enterprise_dashboard_base_path = undef,
-  $enterprise_dashboard_host      = undef,
-  $enterprise_dashboard_port      = undef,
-  $enterprise_dashboard_refresh   = undef,
-  $enterprise_dashboard_user      = undef,
-  $enterprise_dashboard_pass      = undef,
-  $enterprise_dashboard_ssl       = undef,
-  $enterprise_dashboard_audit     = undef,
-  $enterprise_dashboard_github    = undef,
-  $enterprise_dashboard_gitlab    = undef,
-  $enterprise_dashboard_ldap      = undef,
-  $path                           = '$PATH',
-  $redact                         = undef,
-  $deregister_on_stop             = false,
-  $deregister_handler             = undef,
-  $package_checksum               = undef,
-  $windows_pkg_url                = undef,
-  $windows_repo_prefix            = 'https://repositories.sensuapp.org/msi',
-  $windows_logrotate              = false,
-  $windows_log_number             = '10',
-  $windows_log_size               = '10240',
-  $windows_package_provider       = undef,
-  $windows_choco_repo             = undef,
-  $windows_package_name           = 'Sensu',
-  $windows_package_title          = 'sensu',
+  Pattern[/^absent$/, /^installed$/, /^latest$/, /^present$/, /^\d[\d\.\-\w]+$/] $sensu_plugin_version = 'installed',
+  Boolean            $install_repo = true,
+  Boolean            $enterprise = false,
+  Pattern[/^absent$/,/^installed$/,/^latest$/,/^present$/,/^[\d\.\-]+$/] $enterprise_version = 'latest',
+  Optional[String]   $enterprise_user = undef,
+  Optional[String]   $enterprise_pass = undef,
+  Boolean            $enterprise_dashboard = false,
+  String             $enterprise_dashboard_version = 'latest',
+  Boolean            $manage_repo = true,
+  Enum['main','unstable'] $repo = 'main',
+  Optional[String]   $repo_source = undef,
+  String             $repo_key_id = 'EE15CFF6AB6E4E290FDAB681A20F259AEB9C94BB',
+  String             $repo_key_source = 'https://sensu.global.ssl.fastly.net/apt/pubkey.gpg',
+  Optional[String]   $repo_release = undef,
+  Variant[Undef,Integer,Pattern[/^(\d+)$/]] $spawn_limit = undef,
+  String             $enterprise_repo_key_id = '910442FF8781AFD0995D14B311AB27E8C3FE3269',
+  Boolean            $client = true,
+  Boolean            $server = false,
+  Boolean            $api = false,
+  Boolean            $manage_services = true,
+  Boolean            $manage_user = true,
+  Boolean            $manage_plugins_dir = true,
+  Boolean            $manage_handlers_dir = true,
+  Boolean            $manage_mutators_dir = true,
+  Variant[Undef,Integer,Pattern[/^(\d+)$/]] $rabbitmq_port = undef,
+  Optional[String]   $rabbitmq_host = undef,
+  Optional[String]   $rabbitmq_user = undef,
+  Optional[Sensitive[String]]   $rabbitmq_password = undef,
```

Comment:
My suggestion is to accept Variant[Sensitive, String].  If we get a String, immediately cast it to a sensitive.  This will maintain backwards compatibility, but will unfortunately require some refactoring to make sure only the sensitive variables are used internally.

Another approach is to just leave it as is and make this a breaking change.  @ghoneycutt Thoughts on if the sensitive parameters should be backwards compatible or not?  I think yes, because of Hiera.